pymeasure/instruments/lecroy/lecroyT3DSO1204.py

```diff
@@ -30,33 +30,36 @@
 import numpy as np
 
 from pymeasure.instruments import Instrument
-from pymeasure.instruments.validators import strict_discrete_set, strict_range
+from pymeasure.instruments.validators import strict_discrete_set, strict_range, \
+    strict_discrete_range
 
 log = logging.getLogger(__name__)
 log.addHandler(logging.NullHandler())
 
 
-def _sanitize_source(source):
+def sanitize_source(source):
     """ Parse source string
 
     :param source can be "cX", "ch X", "chan X", "channel X", "math" or "line", where X is
     a single digit integer. The parser is case and white space insensitive.
     :return: can be "C1", "C2", "C3", "C4", "MATH" or "LINE. """
 
-    match = re.match(r"\s*(C\s*\d|CH\s*\d|CHAN\s*\d|CHANNEL\s*\d|MATH|LINE)", source,
-                     re.IGNORECASE)
+    match = re.match(r"^\s*(C|CH|CHAN|CHANNEL)\s*(?P<number>\d)\s*$|"
+                     r"^\s*(?P<name_only>MATH|LINE)\s*$", source, re.IGNORECASE)
     if match:
-        source = match.group(1).replace(" ", "").upper()
-        if source not in ["MATH", "LINE"]:
-            source = "C" + source[-1]
+        if match.group("number") is not None:
+            source = "C" + match.group("number")
+        else:
+            source = match.group("name_only")
+        source = source.replace(" ", "").upper()
```

Comment:
Do you expect any spaces in the "source" string?

Reply:
Yes there might be spaces, for example "Channel 1"

Reply:
In the input string might be spaces, but you compose `source = "C" + match.group("number")` and neither "C" nor  the group "number" contain spaces. Similarly, `(?P<name_only>MATH|LINE)` returns no spaces.

I see that you have tests in place. Just remove `replace(...)` and run the test, they should pass nevertheless.

Btw, you could test just the function `sanitize` in itself (verifying that the input strings produce the right output string) instead of indirectly testing through the `ch` method of the instrument. (I'm happy though, that it is tested at all)

Reply:
I removed the call to `replace(...)` and I am testing the function `sanitize` directly.